cmake/OpenCVFindLibsGrfmt.cmake

```diff
@@ -17,8 +17,10 @@ else()
     unset(_zlib_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES)
   endif()
   if(ZLIB_FOUND AND ANDROID)
-    if(ZLIB_LIBRARIES MATCHES "/usr/lib.*/libz.so$")
+    if(ZLIB_LIBRARY MATCHES "/usr/lib.*/libz.so$")
+      set(ZLIB_LIBRARY z)
       set(ZLIB_LIBRARIES z)
+      set(ZLIB_LIBRARY_RELEASE z)
```

Comment:
I believe, this should be done by `find_package(ZLIB "${MIN_VER_ZLIB}")`.
IMHO, it is not an OpenCV responsibility trying to fix that.

---

Try to use `cmake ... -trace-expand` (add `-DCMAKE_TRACE_MODE=ON` to bypass very verbose output) to determine which FindZLIB.cmake is used.
Do you use CMake from Android SDK?

Reply:
Hey, thanks for taking a look at this. I'm using cmake 3.22, not from the android sdk. FindZLIB is included as expected from cmake's own modules:

```
/opt/cmake-3.22.1-linux-x86_64/share/cmake-3.22/Modules/FindZLIB.cmake(58): ...
```

One could argue that the above is a buggy interaction between cmake and the android ndk. It affects me, because I build my own, but it doesn't affect the official OpenCV build.

If you do not wish to support this workaround, you can close the PR and bug and I'll just keep a patch locally that fixes the OpenCV sources before we build. That's also fine.

Suggestions are welcome.

Reply:
Prefer using CMake from Android SDK (it is available since NDK r15). This version is assumed to be more stable for Android platform than regular CMake.

Reply:
So, the latest CMake from the Android SDK (3.18) has the same problem. That's no wonder, since FindZLIB hasn't changed materially since quite a long time ago. I'm not sure where you want to take this PR.